src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,69 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOURS_INDEX = 2;
+    private static final int SALARY_INDEX = 3;
+    private static final String REPORT_FORMAT = "%s - %d";
+    private static final String REPORT_BEGINNING = "Report for period ";
+    private static final String REPORT_DASH = " - ";
+    private final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
+    public String getSalaryInfo(String[] names, String[] data,
+                                String dateFrom, String dateTo) {
+        StringBuilder builder = new StringBuilder();
+        LocalDate localDateFrom = parseDate(dateFrom);
+        LocalDate localDateTo = parseDate(dateTo);
+
+        builder.append(REPORT_BEGINNING)
+                .append(dateFrom)
+                .append(REPORT_DASH)
+                .append(dateTo);
+
+        for (String name : names) {
+            int salary = getSalarySumFromData(data, name, localDateFrom, localDateTo);
+            builder.append(System.lineSeparator())
+                    .append(String.format(REPORT_FORMAT, name, salary));
+        }
+        return builder.toString();
+    }
+
+    private int getSalarySumFromData(String[] data, String name,
+                                     LocalDate localDateFrom,
+                                     LocalDate localDateTo) {
+        int salary = 0;
+        for (String line : data) {
+            String[] parsedLine = line.split(" ");
+            if (!name.equals(parsedLine[NAME_INDEX])) {
+                continue;
+            }
+
+            boolean checkResult = checkForDateSegment(parsedLine,
+                    localDateFrom, localDateTo);
+            if (checkResult) {
+                salary += Integer.parseInt(parsedLine[SALARY_INDEX])
+                        * Integer.parseInt(parsedLine[HOURS_INDEX]);
+            }
+        }
+        return salary;
+    }
+
+    private boolean checkForDateSegment(String[] data,
```

Comment:
checkForDateSegment is not a good name for a method that returns a boolean, the method name should be something like a question that can only be answered with yes or no